pkg/service/cluster/service.go

```diff
@@ -98,40 +101,81 @@ func (s *Service) Client(ctx context.Context, clusterID uuid.UUID) (*scyllaclien
 	return s.clientCache.Client(ctx, clusterID)
 }
 
-func (s *Service) client(ctx context.Context, clusterID uuid.UUID) (*scyllaclient.Client, error) {
-	s.logger.Info(ctx, "Creating new Scylla REST client", "cluster_id", clusterID)
+func (s *Service) createClientNoValidation(c *Cluster) (*scyllaclient.Client, error) {
+	config := scyllaclient.DefaultConfigWithTimeout(s.timeoutConfig)
+	config.Hosts = c.KnownHosts
+	if c.Port != 0 {
+		config.Port = fmt.Sprint(c.Port)
+	}
+	config.AuthToken = c.AuthToken
+
+	return scyllaclient.NewClient(config, s.logger.Named("client"))
+}
+
+// createClient creates Scylla API that load balances calls to every node from given cluster.
+// There may be a situation that cluster keeps outdated information about list of available hosts.
+// To work it around:
+//   - function iterates over all currently known hosts
+//   - calls consecutive client to get list of available hosts known by Scylla server
+//   - updates list of known hosts to Scylla Manager DB
+//   - returns client created on top of list of hosts returned by the Scylla server
+func (s *Service) createClient(ctx context.Context, clusterID uuid.UUID) (*scyllaclient.Client, error) {
+	s.logger.Info(ctx, "Creating new Scylla HTTP client", "cluster_id", clusterID)
 
 	c, err := s.GetClusterByID(ctx, clusterID)
 	if err != nil {
 		return nil, err
 	}
 
-	client, err := s.createClient(c)
-	if err != nil {
-		return nil, errors.Wrap(err, "create client")
+	config := scyllaclient.DefaultConfigWithTimeout(s.timeoutConfig)
+	if c.Port != 0 {
+		config.Port = fmt.Sprint(c.Port)
 	}
-	defer client.Close()
+	config.AuthToken = c.AuthToken
+	hostsList := append([]string{c.Host}, c.KnownHosts...)
 
-	hosts, err := s.discoverHosts(ctx, client)
-	if err != nil {
-		return nil, errors.Wrap(err, "discover cluster topology")
-	}
-	if err := s.setKnownHosts(c, hosts); err != nil {
-		return nil, errors.Wrap(err, "update cluster")
+	for _, host := range hostsList {
+		config.Hosts = []string{host}
+
+		knownHosts, err := s.findKnownHosts(ctx, config)
+		if err != nil {
+			s.logger.Error(ctx, "Cannot find known hosts using coordinator host.", "host", host, "error", err)
+			continue
+		}
```

Comment:
I understand that you create 1-host clients in order to ensure that this particular host will be queried for cluster status, but this can be achieved by using the `forceHost` on context passed to the transport. E.g.:
```go
// NodeInfo returns basic information about `host` node.
func (c *Client) NodeInfo(ctx context.Context, host string) (*NodeInfo, error) {
	p := operations.NodeInfoParams{
		Context: forceHost(ctx, host),
	}
```

If that's the case, I would be cleaner/faster to create a client with all hosts (like usual) and then use the `forceHost` to find the node that can be our source of truth for known hosts.

Reply:
makes sense, let me change it

Reply:
Done